recipes/rocksdb/all/conanfile.py

```diff
@@ -97,6 +116,14 @@ def validate(self):
 
         if is_msvc(self) and Version(self.settings.compiler.version) < "191":
             raise ConanInvalidConfiguration("Rocksdb requires MSVC version >= 191")
+        
+        if self.options.shared and self.options.get_safe("with_folly"):
+            # https://github.com/facebook/rocksdb/blob/v10.5.1/CMakeLists.txt#L603
+            raise ConanInvalidConfiguration(f"{self.ref} does not support a shared build with folly")
+
+        if self.settings.compiler in ["apple-clang", "clang"] and self.options.get_safe("with_folly"):
+            # https://github.com/facebook/rocksdb/issues/13895
+            raise ConanInvalidConfiguration(f"{self.ref} does not support folly with {self.settings.compiler}.")
```

Comment:
If that's the problem, why not replace that flag?

```suggestion
```
And let's add this piece of code:

```python
    def build(self):
        apply_conandata_patches(self)
        if self.settings.compiler in ["apple-clang", "clang"] and self.options.get_safe("with_folly"):
            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "-Wl,--copy-dt-needed-entries", "-Wl")
```
This worked fine for me. Wdyt?

Reply:
cc @uilianries